### DISCUSSION TITLE
pre/post save action on insert, update or delete

### DISCUSSION THREAD

**Discussion by Mujhtech** [simolus3/drift#2542] [Q&A]
Please how do you think on can implement a pre/post save action when inserting or updating or deleting an item

Use case: for like custom validation from a json

**Comment by simolus3** [comment]
Drift doesn't have support for hooks like that. You could build something like this by writing a [trigger](http://sqlite.org/lang_createtrigger.html) in a [drift file](https://drift.simonbinder.eu/docs/using-sql/drift_files/) that aborts if the validation fails, but that is unlikely to scale.

Another approach could be to use [custom row classes](https://drift.simonbinder.eu/docs/advanced-features/custom_row_classes/) for inserts and make sure the constructor of that class validates the properties.

But perhaps the reasonable solution is to write a wrapping repository around drift that does the validation.
